### ISSUE TITLE
Interface language does not pick system's language

### ISSUE THREAD

**Issue by jotjot** [Komet/MediaElch#1479]
This is not exactly a bug, probably. 
Finally, after a couple of years, I have decided to download the latest version today (on MacOS). And the first thing that happens is that the interface language switches to Portuguese. 
I am in Brazil (Region), that is right, but my language preferences are set to English (first), German (Second) and Portuguese (third). Hence I expected the interface to be in English. I also did not find a preference setting where I could set that interface language. 
Am I missing something? Thank you for your help.

**Comment by ticao2** [comment]
I use Win 11. I don't know how it is for Mac.
A long time ago the language was automatically set using the language of your location, of your country.
Then the feature of setting the language using the `advancedsettings.xml` file was implemented.
https://mediaelch.github.io/mediaelch-doc/settings.html#advanced-settings
Maybe you were using it and don't remember.

**Comment by jotjot** [comment]
Thank you for the hint. I never installed it on this computer and I just verified that there is no advancedsettings.xml file in the ApplicationSupport folder for MediaElch.

**Comment by bugwelle** [comment]
Hi,

the `advancedsettings.xml` language setting is a workaround which I use for testing (because my system is German, but I test using English language). 

Unless you already had such file, MediaElch uses [QLocale::system()](https://doc.qt.io/qt-5/qlocale.html#system) to determine the language. 

To debug this, could you go to "About MediaElch" -> "Developer" and tell me what "Default UI languages" says?
See the screenshot below for an example in German (my system):

![Screenshot_20221009_141213](https://user-images.githubusercontent.com/1667306/194756203-549f467b-d4cb-4d05-9747-1cce484389a7.png)

Regards,
Andre

**Comment by jotjot** [comment]
Hi Andre!
Here you are the screenshots of MediaElch and the system settings.
![Screen Shot 2022-10-09 at 10 47 16](https://user-images.githubusercontent.com/2178333/194760540-930869c3-984c-4df3-b59d-8595983798ed.png)
![Screen Shot 2022-10-09 at 10 47 35](https://user-images.githubusercontent.com/2178333/194760545-3ad6d218-f787-495a-a8b6-c2d54fa5662d.png)

**Comment by bugwelle** [comment]
Thanks! Yes, `en-PR` should  be "English".  We use Qt's locale support. There are and were issues with it. It's also why we don't have a `en-US` translation (only the default `en`).

I saw https://github.com/Komet/MediaElch/issues/1002 again. That's exactly your issue! The blog post at https://www.kdab.com/fixing-a-common-antipattern-when-loading-translations-in-qt/ explains what we did (and apparently still do) wrong.

I will look into it. Thank you very much for reporting this issue and reminding me that it still exists!

This is a duplicate of https://github.com/Komet/MediaElch/issues/1002 but I will keep it open until this issue is resolved.

-------------

Note to self: Test by changing the language on my MacBook as jotjot did.

**Comment by jotjot** [comment]
Thank you for your return. I did not realize that this was a duplicate, as I did not see the origin of the problem. 
I am looking forward for a solution.

**Comment by bugwelle** [comment]
> I did not realize that this was a duplicate, as I did not see the origin of the problem.

No worries, I did not see the original issue as well. :-)

--------------

Interestingly, it works for German (`en-DE`), but if I add Portuguese __and__ set my location to Brazil, it suddenly prefers pt-BR.

I've played around and it turns out: This only happens if Qt finds a language "that fits better". For Portuguese, we have `pt-BR` and `pt-PT`. For all other languages, we don't have a region encoded, only `de` (instead of `de-DE`). If I now add a language `de-DE`, suddenly MediaElch is in German on my system instead of the preferred UI language...

Now that I look at the blog-post again, I even commented on it!

> Hi,
> 
> IMHO, there is a huge pitfall because the documentation is wrong regarding the order of locales it loads.
> If you look at https://doc.qt.io/qt-5/qtranslator.html#load-1 the order for (fr-CA, de) is
> 
> 1. fr_CA
> 2. fr
> 3. de
> 
> But in reality and according to the source comments at https://github.com/qt/qtbase/blame/40143c189b7c1bf3c2058b77d00ea5c4e3be8b28/src/corelib/kernel/qtranslator.cpp#L757 the order is:
> 
> 1. fr_CA
> 2. de
> 3. fr
> 
> This was a huge pitfall for me because on a fresh Windows 10 Education installation for Germany, QLocale::system().uiLanguages() returns (“de-DE”, “en-US”). And my application had “App_de.qm” and “App_en_US.qm” > files. The English version was loaded _all_ the time!
> 
> Is there anyway to first try to load all German versions and only then use the second UI Language?
> 
> Regards,
> Andre

We use Qt 5.15.2 for our macOS releases. The fix is only available in 6.0.1 and later: https://bugreports.qt.io/browse/QTBUG-86179

Long story short --> nothing I can (easily) do.  We rely on pre-build Qt releases from the Qt company.

I can't update to Qt 6, yet, because we need to support macOS 10.13.

However, it may be a good idea to _also_ provide Qt6 releases for macOS and explicitly mention that it requires macOS 11 or later.

I'm closing this issue because from MediaElch's side, I can't fix it. I will comment on https://github.com/Komet/MediaElch/issues/1002 when macOS releases for Qt6 are available.

**Comment by bugwelle** [comment]
FYI, we now have MediaElch snapshots with Qt6 for macOS 11 and later. :)
https://mediaelch-downloads.ameyering.de/snapshots/macOS_11_or_later/

Also for Windows:
https://mediaelch-downloads.ameyering.de/snapshots/Windows_10_or_later/
